lib/oxide-barcode/src/lib.rs

```diff
@@ -42,22 +114,97 @@ pub enum ParseError {
     SerializedSize,
 )]
 #[repr(C, packed)]
-pub struct VpdIdentity {
+pub struct OxideIdentity {
     pub part_number: [u8; Self::PART_NUMBER_LEN],
     pub revision: u32,
     pub serial: [u8; Self::SERIAL_LEN],
 }
 
-impl VpdIdentity {
+impl OxideIdentity {
     pub const PART_NUMBER_LEN: usize = 11;
     pub const SERIAL_LEN: usize = 11;
-}
+    const OXV2: &'static [u8] = b"0XV2:";
+
+    /// Maximum length in bytes of a barcode string.
+    pub const MAX_LEN: usize =
+        Self::OXV2.len() + Self::PART_NUMBER_LEN + Self::SERIAL_LEN
+        + 3 // revision part
+        + 2 // delimiters
+        ;
+
+    /// Encodes this VPD identity into `buf` in the [`0XV2`] format.
+    ///
+    /// If this identity was parsed from an `0XV1`-formatted barcode string, it
+    /// will be normalized to an `0XV2`-formatted string. If it was originally
+    /// encoded as `0XV2`, the output should be identical to the original
+    /// barcode.
+    ///
+    /// [`0XV2`]: https://rfd.shared.oxide.computer/rfd/0308#_0xv2
+    pub fn encode_oxv2(&self, buf: &mut [u8]) -> Result<usize, EncodeError> {
+        fn write_chunk(offset: &mut usize, buf: &mut [u8], data: &[u8]) {
+            // if a serial or part number was shorter than 11 characters, it may
+            // be nul-padded. handle that by chopping off any nuls.
+            let len =
+                data.iter().position(|b| *b == b'\0').unwrap_or(data.len());
+            buf[*offset..*offset + len].copy_from_slice(&data[..len]);
```

Comment:
I have a slight preference for
```suggestion
            buf[*offset..][..len].copy_from_slice(&data[..len]);
```

Reply:
TIL you can do that, that's cute!

My one concern is that it might generate two panic sites (since the two indexing expressions will have separate source positions), but I dunno if that matters all that much.